src/dbnode/storage/index/results.go

```diff
@@ -48,6 +48,10 @@ type results struct {
 	resultsMap     *ResultsMap
 	totalDocsCount int
 
+	// Utilization stats, do not reset:
+	maxObservedSize        int
```

Comment:
I'd probably name this `resultsMapCapacity` - too bad there is no `cap(map)`...

Reply:
And maybe the other one `consecutiveTimesUnderCapacity`

Reply:
👍